Source/ChocolateyGui/ChocolateyExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Chocolatey" file="ChocolateyExtensions.cs">
+//   Copyright 2014 - Present Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ChocolateyGui
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using chocolatey;
+    using chocolatey.infrastructure.results;
+
+    public static class ChocolateyExtensions
+    {
+        private static readonly ConcurrentDictionary<string, CancellationTokenSource> _cancellationTokenSources =
+            new ConcurrentDictionary<string, CancellationTokenSource>(StringComparer.OrdinalIgnoreCase);
+
+        public static Task RunAsync(this GetChocolatey chocolatey)
+        {
+            return Task.Run(() => chocolatey.Run());
+        }
+
+        public static Task<ICollection<T>> ListAsync<T>(this GetChocolatey chocolatey)
+        {
+            return Task.Run(() => (ICollection<T>)chocolatey.List<T>().ToList());

```

Comment:
not really related to this change (as it's just been moved..) but there is a potentail crash/ uncaught exception:
![image](https://user-images.githubusercontent.com/5354972/30503138-6801453c-9a69-11e7-8abf-5f6169dfb3bf.png)

Reply:
Oh joy...

Reply:
nevermind, all good
- just set my VisualStudio to go crazy on exceptions :-) (sorry for that ^^)